includes/theme-support/class-wc-unsupported-theme-product-single.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Unsupported theme single product page enhancements.
+ *
+ * @package  WooCommerce/Classes
+ * @since   3.3.0
+ * @version 3.3.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Single product pages support for themes that don't declare WooCommerce support.
+ *
+ * @class   WC_Unsupported_Theme_Product_Single
+ * @package WooCommerce/Classes
+ * @since   3.3.0
+ */
+class WC_Unsupported_Theme_Product_Single {
```

Comment:
Thinking about this in relation to my archive PR; do we have a separate class for each, a class combined for both, or use the template loader like I did?

Reply:
I'm going to move it into the template loader. The whole class is really only a couple methods.